pkg/utils/apiserver.go

```diff
@@ -0,0 +1,31 @@
+package utils
+
+import (
+	"context"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GetClusterAPIServerInfo returns the cluster API server URL
+// In Hypershift API Server and internal API Server are not required to be in a subdomain of the base domain.
+//
+// For example, given the base domain `openshift.example.com`,
+// an API server could be in `api.hypershift.local`.
+//
+// It uses this API https://docs.openshift.com/container-platform/4.7/rest_api/config_apis/dns-config-openshift-io-v1.html
+func GetClusterAPIServerInfo(ctx context.Context, configClient configv1client.ConfigV1Interface) ([]string, error) {
+	infra, err := configClient.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return []string{}, err
+	}
+
+	urls := []string{}
+	for _, url := range []string{infra.Status.APIServerURL, infra.Status.APIServerInternalURL} {
```

Comment:
This will cause panic if the state is not set (probably an edge case).

Reply:
I'm going to add some unit tests there and refactor the logic. Thanks for the review.

Reply:
Solved in 1e7431666a292f50e34e76fa700279364f700078